internal/bminventory/inventory.go

```diff
@@ -910,8 +909,8 @@ func (b *bareMetalInventory) GenerateClusterISOInternal(ctx context.Context, par
 		return b.GetClusterInternal(ctx, installer.GetClusterParams{ClusterID: *cluster.ID})
 	}
 
-	ignitionConfig, formatErr := b.IgnitionBuilder.FormatDiscoveryIgnitionFile(cluster,
-		b.IgnitionConfig, params.ImageCreateParams, false, b.authHandler.AuthType())
+	cluster.ImageInfo.Type = params.ImageCreateParams.ImageType
```

Comment:
Maybe this should be set with the rest of the image info before the update?
Otherwise I feel like it needs a comment here to explain what is going on.

I'm worried that we'll add something that needs to be set for `FormatDiscoveryIgnitionFile` and forget that we might need to configure it manually here rather than it being set in the cluster record.

Reply:
Comment works for me

Reply:
I've added a comment to describe this line.
Even today some of the parameters are being stored to the DB prior to the image generation and some of them only after a successful generation of them.

The alternative would be exposing another variable `imageType` to `FormatDiscoveryIgnitionFile` instead of relying on the cluster to carry it.
However, note that GET to the discovery-ignition doesn't allow to provide any value when the request is submitted, and it used to ignore that imagetype.

Reply:
In one of the drafts of https://github.com/openshift/assisted-service/pull/1242 I proposed a pattern similar to Java's Builder pattern (but using functional parameters which is GO's style) - that pattern would create the ImageBuilder using a fluent calls `ImageBuilder().Cluster(c).ImageType(t).SafeForLogs(true).Config(cfg).Build()` which would be more clear about the expected parameters and allow to validate each of them individually. 

However that draft was abandoned in favor of the last one which is more conventional.